gen/generate_bindings.jl

```diff
@@ -0,0 +1,226 @@
+
+Find out why `using SimpleDirectMediaLayer` is slow:
+
+    `julia -e using pkgname`
+
+    time(s) pkgname
+    6.7     SimpleDirectMediaLayer
+    6.6     SDL2_jll, SDL2_image_jll, SDL2_mixer_jll, SDL2_ttf_jll, ColorTypes
+    5.7     SDL2_jll, SDL2_image_jll, SDL2_mixer_jll, SDL2_ttf_jll
+    3.5     SDL2_jll
+    4.5     SDL2_image_jll
+    1.2     ColorTypes
+
+    3.0     Strs
+    1.5     ThreadsX
+
+    Suggests we can only save time by loading the jlls lazily or by modifying
+    the dependencies directly.
+
+
+### How to integrate manual changes
+
+e.g. I'd want to edit SDL_Event and rename WindowShowMode
+
+If the diffs are stable enough across versions I could just manually merge. Let's test that...
+    Functions move up and down in an unpredictable way between 2.0.5 and 2.0.12, so lexical sorting at least within each file would probably be sensible.
+
+
+### SDL_ prefix issue
+
+Literally every function has an SDL_ prefix, so we can remove those fine.
+
+With the structures and constants, however, there's a mix and we'll get (rare)
+collisions if we just automatically remove all prefixes (WindowShowMode and
+SDL_WindowShowMode are both defined, for example).
+
+
+### Low priority aesthetic issues
+
+Some types have lowercase names (mutex, sem, semaphore) (Maybe shouldn't fix this?)
+
+Some types are aliased (Uint64 - UInt64, sem - semaphore, ...)
+
+There are docstrings in the headers that we could extract. This site has them, but it's not very ergonomic. We could extract them with doxygen or whatever.
+https://happiness_follows.gitee.io/doc_translate/SDL2-2.0.7/html/annotated.html
+
+"""
+
+const dir = @__DIR__
+cd(dir)
+
+# Latest
+# sdl2_version = "2.0.12"
+
+# The version in the comment of SDL.jl
+sdl2_version = "2.0.5"
+
+urls = [
+    :sdl2 => "http://www.libsdl.org/release/SDL2-$sdl2_version.tar.gz"
+    # :sdl2_image => "https://www.libsdl.org/projects/SDL_image/release/SDL2_image-2.0.5.tar.gz"
+    # :sdl2_mixer => "https://www.libsdl.org/projects/SDL_mixer/release/SDL2_mixer-2.0.4.tar.gz"
+    # :sdl2_ttf => "https://www.libsdl.org/projects/SDL_ttf/release/SDL2_ttf-2.0.15.tar.gz"
+]
+
+# Get sources
+mkpath("downloads")
+for (_, url) in urls
+    if !isfile(joinpath("downloads", basename(url)))
+        download(url, joinpath("downloads", basename(url)))
+    end
+end
+
+# Unpack their include dirs
+using DataDeps: unpack
+
+# Let's just do this one at a time for now.
+
+isdir("SDL2-$sdl2_version") ||
+    unpack("downloads/SDL2-$sdl2_version.tar.gz"; keep_originals=true)
+
+
+### Generate bindings
+
+# This is copied from the Clang.jl readme and then I changed the obvious bits.
+using Clang: CLANG_INCLUDE, InternalOptions, init
+
+# On windows, Clang will complain that it can't find a load of headers and this
+# genuinely causes problems. Set a proper include or compile on a system where
+# clang is more likely to find the headers.
+LIBSDL_INCLUDE = "SDL2-$sdl2_version/include/"
```

Comment:
Can't you use the header files in `joinpath(SDL2_jll.artifact_dir, "include")` instead of downloading the source code again?

Reply:
Maybe! I'll have a look.

Reply:
I want control at the moment over which version I'm using because I want to see how hard diffs are going to be between generated outputs for different SDL versions.